### DISCUSSION TITLE
Time taken to import the library

### DISCUSSION THREAD

**Discussion by husqD** [aquacropos/aquacrop#8] [Q&A] [answered]
Hi,
Why the library take like 30 seconds to import?
I have tried importing only what I need, but the library takes the same time to initialize.
Why is this happening?
Thank you

**Comment by thomasdkelly** [comment, marked as answer]
Hi I use numba jit compilation to speed up the simulation and for that it needs to compile the model functions on the first run through. In the aquacrop __init__.py I run a test simulation so that this compilation occurs when the user first imports the module rather than when they run their first simulation.

Although not ideal I decided the trade offs of numba jit compilation were better than either writing the base code in cython or having each simulation take 5x slower in pure python/numpy. 


Tom

**Reply by husqD** [reply]
Okey, thank you for your response.
